core/src/main/scala/org/apache/spark/status/api/v1/StagesResource.scala

```diff
@@ -40,8 +43,27 @@ private[v1] class StagesResource extends BaseAppResource {
       @PathParam("stageId") stageId: Int,
       @QueryParam("details") @DefaultValue("true") details: Boolean): Seq[StageData] = {
     withUI { ui =>
-      val ret = ui.store.stageData(stageId, details = details)
+      var ret = ui.store.stageData(stageId, details = details)
       if (ret.nonEmpty) {
+        for (r <- ret) {
+          val executorIdArray = r.executorSummary.get.keys.toArray
```

Comment:
remove executorIdArray and just put r.executorSummary.get.keys.toArray inside for